jetty-http/src/main/java/org/eclipse/jetty/http/pathmap/RegexPathSpec.java

```diff
@@ -357,78 +371,92 @@ private class RegexMatchedPath implements MatchedPath
 
             if (groupCount == 0)
             {
                 pathMatch = path;
                 pathInfo = null;
+                return;
             }
-            else if (groupCount == 1)
+
+            if (groupCount == 1)
             {
-                if (idxNameStart >= 0)
+                // we know we are splitting
+                int idxNameEnd = endOf(matcher, "name");
+                if (idxNameEnd >= 0)
                 {
-                    pathMatch = path.substring(idxNameStart, idxNameEnd);
+                    pathMatch = path.substring(0, idxNameEnd);
                     pathInfo = path.substring(idxNameEnd);
-                }
-                else
-                {
-                    pathMatch = path.substring(0, matcher.start(1));
-                    pathInfo = matcher.group(1);
-                }
-            }
-            else
-            {
-                if (idxNameStart >= 0)
-                {
-                    pathMatch = path.substring(idxNameStart, idxNameEnd);
-                    if (idxInfoStart >= 0)
-                    {
-                        pathInfo = path.substring(idxInfoStart, idxInfoEnd);
-                    }
-                    else
+
+                    // If split on pathMatch ends with '/'
+                    // AND pathInfo doesn't have one, move the slash to pathInfo only move 1 level
+                    if (pathMatch.length() > 0 && pathMatch.charAt(pathMatch.length() - 1) == '/' &&
+                        !pathInfo.startsWith("/"))
                     {
-                        pathInfo = path.substring(idxNameEnd);
+                        pathMatch = pathMatch.substring(0, pathMatch.length() - 1);
+                        pathInfo = '/' + pathInfo;
                     }
+                    return;
                 }
-                else if (idxInfoStart >= 0)
-                {
-                    pathMatch = path.substring(0, idxInfoStart);
-                    pathInfo = path.substring(idxInfoStart, idxInfoEnd);
-                }
-                else
+
+                // Use start of anonymous group
+                int idx = matcher.start(1);
+                if (idx >= 0)
                 {
-                    pathMatch = path;
-                    pathInfo = null;
+                    pathMatch = path.substring(0, idx);
+                    pathInfo = path.substring(idx);
+
+                    if (pathMatch.length() > 0 && pathMatch.charAt(pathMatch.length() - 1) == '/' &&
+                        !pathInfo.startsWith("/"))
+                    {
+                        pathMatch = pathMatch.substring(0, pathMatch.length() - 1);
+                        pathInfo = '/' + pathInfo;
+                    }
+                    return;
                 }
             }
+
+            // Reach here we have 2+ groups
+
+            String gName = valueOf(matcher, "name");
+            String gInfo = valueOf(matcher, "info");
```

Comment:
Two exceptions may be created and ignored here.  So again, I think we need a TODO to optimise that away in future.